### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by shabtaisharon**
* Support uploadNewFilesOnly on Archives
* Support ignoreDuplicates on Restores
* More testing

### CODE REVIEW

File: SpectraLogic.SpectraRioBrokerClient.Integration.Test/SpectraRioBrokerClientErrorTests.cs [hunk 1]
```diff
@@ -516,6 +528,82 @@ public void RestoreErrorTests()
...
+                        });
+                        SpectraRioBrokerClientFixture.SpectraRioBrokerClient.Restore(restoreRequest);
+                        Assert.Fail();
+                    },
+                    new List<UnprocessableError>
+                    {
+                        new UnprocessableError(fileName1, "file", "duplicate_file_brokers"),
+                    });
+
+                var deleteF1Request = new DeleteFileRequest(SpectraRioBrokerClientFixture.BrokerName, fileName1);
```
Comment (RachelTucker) [inline SpectraLogic.SpectraRioBrokerClient.Integration.Test/SpectraRioBrokerClientErrorTests.cs]: if your reusing this var for deleting both, is there any reason to namespace it with F1?

File: SpectraLogic.SpectraRioBrokerClient.Integration.Test/SpectraRioBrokerClientFixture.cs [hunk 2]
```diff
@@ -28,7 +28,9 @@ public static class SpectraRioBrokerClientFixture
         #region Public Fields
 
         public static readonly string AgentName = "bp_agent";
+        //public static readonly string AgentName2 = "bp_agent_2";
```
Comment (RachelTucker) [inline SpectraLogic.SpectraRioBrokerClient.Integration.Test/SpectraRioBrokerClientFixture.cs]: any reason not to delete this?

File: SpectraLogic.SpectraRioBrokerClient.Integration.Test/SpectraRioBrokerClientIntegrationTests.cs [hunk 3]
```diff
@@ -129,7 +129,48 @@ public void ArchiveAndRestore()
             }
         }
 
+        [Test, Ignore("https://jira.spectralogic.com/browse/ESCP-371")]
+        public void ArchiveNewFilesOnlyTest_1()
```
Comment (RachelTucker) [inline SpectraLogic.SpectraRioBrokerClient.Integration.Test/SpectraRioBrokerClientIntegrationTests.cs]: I know this test is being ignored, but what is it testing? There are no asserts.

Comment (RachelTucker) [reply SpectraLogic.SpectraRioBrokerClient.Integration.Test/SpectraRioBrokerClientIntegrationTests.cs]: nm, I get it now

File: SpectraLogic.SpectraRioBrokerClient.Integration.Test/SpectraRioBrokerClientErrorTests.cs (hunk 1 above)
Comment (shabtaisharon) [reply SpectraLogic.SpectraRioBrokerClient.Integration.Test/SpectraRioBrokerClientErrorTests.cs]: yep, both requests are deleting F1

File: SpectraLogic.SpectraRioBrokerClient.Integration.Test/SpectraRioBrokerClientIntegrationTests.cs (hunk 3 above)
Comment (shabtaisharon) [reply SpectraLogic.SpectraRioBrokerClient.Integration.Test/SpectraRioBrokerClientIntegrationTests.cs]: 👍

**Pull request merged**